docs/getting-started-tutorial/4-css-selector-transforms.adoc

```diff
@@ -0,0 +1,171 @@
+    "li *" #> messageEntries
+  }
+...
+```
+
+Let's recompile and restart the server:
+
+```
+> container:stop
+[info] stopped o.e.j.w.WebAppContext{/,[file:/Users/Shadowfiend/github/lift-example/src/main/webapp/]}
+[success] Total time: 0 s, completed Oct 6, 2013 2:36:48 PM
+> container:start
+[info] Compiling 1 Scala source to /Users/Shadowfiend/github/lift-example/target/scala-2.9.2/classes...
+[info] jetty-8.1.7.v20120910
+[info] NO JSP Support for /, did not find org.apache.jasper.servlet.JspServlet
+[info] started o.e.j.w.WebAppContext{/,[file:/Users/Shadowfiend/github/lift-example/src/main/webapp/]}
+[info] started o.e.j.w.WebAppContext{/,[file:/Users/Shadowfiend/github/lift-example/src/main/webapp/]}
+[info] Started SelectChannelConnector@0.0.0.0:8080
+```
+
+Now if you pull up the page you'll see something that doesn't look quite right.
+The markup we're producing should look something like:
+
+```
+      <li>It is now 13:25 UTC</li>
+      <li>It is now 13:25 UTC</li>
+      <li>It is now 13:25 UTC</li>
+      <li>It is now 13:25 UTC</li>
+```
+
+If you reload the page, you'll get something like:
+
+```
+      <li>It is now 13:25 UTC</li>
+      <li>It is now 13:25 UTC</li>
+      <li>It is now 13:25 UTC</li>
+      <li>It is now 13:25 UTC</li>
+      <li>It is now 13:26 UTC</li>
+      <li>It is now 13:26 UTC</li>
+      <li>It is now 13:26 UTC</li>
+      <li>It is now 13:26 UTC</li>
+```
+
+What's causing all the repetition? Well, remember when we described what the
+CSS Selector Transform was doing, we said we “select all ``li``s”. We also said
+that the list on the right side means “Lift repeats the matched element **or
+elements**”. So we select all the ``li``s, but in the template there are 4, so
+that the template when viewed alone (say, for a user test, or when a frontend
+developer is editing it) has some content in it. How do we bridge the two
+without getting nasty in our HTML?
+
+Lift lets us tag the extra elements with a class `clearable`:
+
+```
+...
+      <li>Hi!</li>
+      <li class="clearable">Oh, hey there.</li>
+      <li class="clearable">How are you?</li>
+      <li class="clearable">Good, you?</li>
+...
+```
+
+Then, in our snippet, we can use a special transform called `ClearClearable`,
+which will remove all of the tagged elements before we start transforming the
+template:
+
+```
+...
+  def messages = {
+    messageEntries :+= "It is now " + formattedTimeNow
+
+    ClearClearable &
+    "li *" #> messageEntries
+  }
+...
+```
+
+Notice that we combine the two CSS selector transforms here by using `&`. You
+can chain together as many CSS selector transforms as you want this way, as long
+as they don't modify the same parts of the same element. We'll deal with that
+limitation link:13-who-knows[a little later] footnote:[This is because CSS
+selector transforms are optimized for speed, and pass through the nodes a
+single time to make all of the transformations happen.].
```

Comment:
It's probably also worth adding a footnote here regarding indentation rules, since any Scala developer passing through will likely wonder why we don't represent this as:

```scala
ClearClearable &
  "li *" #> messageEntries
```

Reply:
I'm going to punt on this… Not sure that note really belongs in a beginner tutorial. I've also been rethinking the general principle, although I like the look of same-indentation. Mostly because if you consider auto-indentation tools your friend, they can actually catch the common error of forgetting an `&`... Anyway, I don't have a strong thought there for now, but I think we can get away with not mentioning it in the tutorial.